src/crimson/common/config_proxy.h

```diff
@@ -34,25 +35,33 @@ class ConfigProxy : public seastar::peering_sharded_service<ConfigProxy>
   }
 
   // apply changes to all shards
-  // @param func a functor which accepts @c "ConfigValues&", and returns
-  //             a boolean indicating if the values is changed or not
+  // @param func a functor which accepts @c "ConfigValues&"
```

Comment:
i liked the idea that `func()` could return false to indicate that nothing changed, so we could avoid distributing it to other shards. do you think that will just be too rare to be worth special handling?

Reply:
i think there is good chance that user inject/set invalid args to OSD, but in that case `func()` would throw an `invalid_argument` exception. and i assume that whoever calls `set_val()` or `injectargs()` (not yet implemented), will catch this exception and show user proper error messages.

Reply:
or, do you think we should not use exception for handling something not very rare, or something not completely unexpected?